drivers/input/input_ili2132a.c

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (c) 2024 Joel Jaldemark
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT ilitek_ili2132a
+
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/input/input.h>
+#include <zephyr/pm/device.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(ili2132a, CONFIG_INPUT_LOG_LEVEL);
+
+struct ili2132a_data {
+    const struct device *dev;
+    struct gpio_callback gpio_cb;
+    struct k_work work;
+};
+
+struct ili2132a_config {
+    struct i2c_dt_spec bus;
+    struct gpio_dt_spec rst;
+    struct gpio_dt_spec irq;
+};
+
+static void gpio_isr(const struct device *dev, struct gpio_callback *cb,
+                     uint32_t pin)
+{
+    struct ili2132a_data *data = CONTAINER_OF(cb, struct ili2132a_data, gpio_cb);
+    k_work_submit(&data->work);
+}
+
+static void ili2132a_process(const struct device *dev)
+{
+    const struct ili2132a_config *dev_cfg = dev->config;
+    uint8_t buf[8];
+    uint16_t x, y;
+
+    i2c_read_dt(&dev_cfg->bus, buf, 8);
```

Comment:
```suggestion
    i2c_read_dt(&dev_cfg->bus, buf, sizeof(buf));
```

Also should this be a `i2c_burst_read_dt` with the address of the initial register?

Reply:
A multi-byte read after the module issues an interrupt gives what they call report data which includes coordinates and point-ID(0x40 + number of points - 1). So no need to specify a register address.

Reply:
sounds good, just checking